src/app/models/weather-alert.model.ts

```diff
@@ -1,42 +1,32 @@
-  readonly activeAlertId: string;
-  readonly activeAlertInstruction: string;
-  readonly showAlertToast: boolean;
+export class WeatherAlertResponse extends ApiResponseModel {
+  public allActiveWeatherAlerts: WeatherAlertInfo[];
+  public activeAlertDescription: string;
+  public activeAlertEvent: string;
+  public activeAlertHeadline: string;
+  public activeAlertId: string;
+  public activeAlertInstruction: string;
+  public showAlertToast: boolean;
 
-  constructor(apiResponse: any) {
-    try {
-      if (apiResponse && Array.isArray(apiResponse.features)) {
-        this.flowStatus = 'SUCCESS';
-        this.flowStatusMessage = apiResponse.features.length > 0 ? apiResponse.title : 'No active weather alerts in your area.';
-      } else {
-        this.flowStatus = 'FAILURE';
-        this.flowStatusMessage = 'Unable to parse api response';
-      }
-      if (this.flowStatus === 'SUCCESS') {
-        const tempAllActiveWeatherAlerts: WeatherAlertInfo[] = this.parseApiResponse(apiResponse);
-        this.allActiveWeatherAlerts = this.filterDuplicateAlerts(tempAllActiveWeatherAlerts);
-        // the below variables are set to 0 index of array because that will be up-to-date active alert
-        this.activeAlertHeadline = this.allActiveWeatherAlerts && this.allActiveWeatherAlerts[0] && this.allActiveWeatherAlerts[0].headline ? this.allActiveWeatherAlerts[0].headline : '';
-        this.activeAlertDescription = this.allActiveWeatherAlerts && this.allActiveWeatherAlerts[0] && this.allActiveWeatherAlerts[0].description ? this.allActiveWeatherAlerts[0].description : '';
-        this.activeAlertEvent = this.allActiveWeatherAlerts && this.allActiveWeatherAlerts[0] && this.allActiveWeatherAlerts[0].event ? this.allActiveWeatherAlerts[0].event : '';
-        this.activeAlertId = this.allActiveWeatherAlerts && this.allActiveWeatherAlerts[0] && this.allActiveWeatherAlerts[0].id ? this.allActiveWeatherAlerts[0].id : '';
-        this.activeAlertInstruction = this.allActiveWeatherAlerts && this.allActiveWeatherAlerts[0] && this.allActiveWeatherAlerts[0].instruction ? this.allActiveWeatherAlerts[0].instruction : '';
-        this.showAlertToast = Array.isArray(this.allActiveWeatherAlerts) && this.allActiveWeatherAlerts.length > 0;
-      }
-    } catch (e) {
-      this.flowStatus = 'FAILURE';
-      this.flowStatusMessage = e;
-    }
+  constructor(response: any) {
+    super(response);
+    this.processResponse(response);
+  }
+
+  protected processResponse(response: any): void {
+    const tempAllActiveWeatherAlerts: WeatherAlertInfo[] = this.parseApiResponse(response);
+    this.allActiveWeatherAlerts = this.filterDuplicateAlerts(tempAllActiveWeatherAlerts);
+    // the below variables are set to 0 index of array because that will be up-to-date active alert
+    this.activeAlertHeadline = this.allActiveWeatherAlerts && this.allActiveWeatherAlerts[0] && this.allActiveWeatherAlerts[0].headline ? this.allActiveWeatherAlerts[0].headline : '';
```

Comment:
```suggestion
    this.activeAlertHeadline = Array.isArray(this.allActiveWeatherAlerts) && this.allActiveWeatherAlerts[0]?.headline ? this.allActiveWeatherAlerts[0].headline : '';
```
you could refactor all to use optional chaining now 

**atlas standards allow modern JS now...so use of `?.`, `??` and `||` for truthy checking is okay